src/psyclone/tests/psyir/transformations/parallel_loop_trans_test.py

```diff
@@ -530,3 +527,94 @@ def test_paralooptrans_apply(fortran_reader):
     trans = ParaTrans()
     trans.apply(loop, {"force": True})
     assert isinstance(loop.parent.parent, OMPParallelDoDirective)
+
+
+def test_paralooptranas_with_array_privatisation(fortran_reader,
+                                                 fortran_writer):
+    '''
+    Check that the 'privatise_arrays' transformation option allows to ignore
+    write-write dependencies by setting the associated variable as 'private'
+    '''
+    psyir = fortran_reader.psyir_from_source('''
+        subroutine my_sub()
+          integer ji, jj
+          real :: var1(10,10)
+          real :: ztmp(10)
+          real :: ztmp2(10)
+          var1 = 1.0
+          ztmp2 = 1.0
+
+          do ji = 1, 10
+            do jj = 1, 10
+              if (jj == 4) then
+                ztmp2(jj) = 4
+              end if  ! the rest get the value from before the loop
+              ztmp(jj) = var1(ji, jj) + ztmp2(jj)
+            end do
+            do jj = 1, 10
+              var1(ji, jj) = ztmp(jj) * 2
+            end do
+          end do
+        end subroutine my_sub''')
+
+    loop = psyir.walk(Loop)[0]
+    trans = ParaTrans()
+
+    # By default this can not be parallelised because 'ztmp' is shared
+    with pytest.raises(TransformationError) as err:
+        trans.apply(loop)
+    assert "ztmp(jj)\' causes a write-write race condition." in str(err.value)
+    assert "ztmp2(jj)\' causes a write-write race condition." in str(err.value)
+
+    # Now enable array privatisation
+    trans.apply(loop, {"privatise_arrays": True})
+    assert ("!$omp parallel do default(shared), private(ji,jj,ztmp), "
+            "firstprivate(ztmp2)" in fortran_writer(psyir))
+
+    # If the array is accessed after the loop, or is a not an automatic
+    # interface, or is not a plain array, the privatisation will fail
+    psyir = fortran_reader.psyir_from_source('''
+        subroutine my_sub()
+          use other, only: mystruct
+          integer ji, jj
+          real :: var1(10,10)
+          real, save :: ztmp1(10)
```

Comment:
I don't think this one is included in the test code and it should be?

Reply:
No, removed